zeppelin-web-angular/src/app/services/angular-drag-drop.service.ts

```diff
@@ -0,0 +1,328 @@
+          const args = [event, ui].concat(objExtract.args);
+
+          // call either $scope method or constructor's method
+          const targetScope = scope[callback] ? scope : scope[constructor];
+          const targetCallback = scope[callback] || scope[constructor][callback];
+
+          if (typeof targetCallback === 'function') {
+            return targetCallback.apply(targetScope, args);
+          }
+
+          function extract(_callbackName: string) {
+            const atStartBracket =
+              _callbackName.indexOf('(') !== -1 ? _callbackName.indexOf('(') : _callbackName.length;
+            const atEndBracket =
+              _callbackName.lastIndexOf(')') !== -1 ? _callbackName.lastIndexOf(')') : _callbackName.length;
+            const argsString = _callbackName.substring(atStartBracket + 1, atEndBracket);
+            let _constructor =
+              _callbackName.indexOf('.') !== -1 ? _callbackName.substr(0, _callbackName.indexOf('.')) : null;
+            _constructor =
+              scope[_constructor] && typeof scope[_constructor].constructor === 'function' ? _constructor : null;
+
+            return {
+              callback: _callbackName.substring((_constructor && _constructor.length + 1) || 0, atStartBracket),
+              // tslint:disable-next-line:no-any
+              args: argsString ? argsString.split(',').map((item: string) => $parse(item.trim())(scope)) : [],
+              constructor: _constructor
+            };
+          }
+        };
+
+        // tslint:disable-next-line:no-any
+        this.updateModel = function(scope: any, modelPath: string, newValue: any, index?: number) {
+          const getter = $parse(modelPath);
+          const setter = getter.assign;
+          const modelValue = getter(scope);
+
+          if (angular.isArray(modelValue)) {
+            if (typeof index === 'number') {
+              modelValue[index] = newValue;
+            } else {
+              modelValue.push(newValue);
+            }
+          } else {
+            if (setter) {
+              setter(scope, newValue);
+            }
+          }
+
+          scope.$apply();
+        };
+
+        // tslint:disable-next-line:no-any
+        this.removeFromModel = function(scope: any, modelPath: string, index: number) {
+          const getter = $parse(modelPath);
+          const modelValue = getter(scope);
+
+          if (angular.isArray(modelValue) && typeof index === 'number') {
+            modelValue.splice(index, 1);
+            scope.$apply();
+          }
+        };
+      }
+    ]);
+
+    // jqyoui-draggable directive
+    // tslint:disable-next-line:no-any
+    module.directive('jqyouiDraggable', [
+      'customDragDropService',
+      // tslint:disable-next-line:no-any
+      function(dragDropService: any) {
+        return {
+          restrict: 'A',
+          // tslint:disable-next-line:no-any
+          link: function(scope: any, element: any, attrs: any) {
```

Comment:
How about removing the explicit `any`? Type inference would work correctly here.